raiden_contracts/tests/test_deprecation_switch.py

```diff
@@ -0,0 +1,243 @@
+    token_network_registry = deploy_contract(
+        web3,
+        administrator,
+        json_contract['abi'],
+        json_contract['bin'],
+        [
+            secret_registry_contract.address,
+            int(web3.version.network),
+            TEST_SETTLE_TIMEOUT_MIN,
+            TEST_SETTLE_TIMEOUT_MAX,
+        ],
+    )
+
+    # Make sure deployer is administrator
+    assert token_network_registry.functions.administrator().call() == administrator
+
+    # Only the administrator can deploy TokenNetwork contracts
+    with pytest.raises(TransactionFailed):
+        token_network_registry.functions.createERC20TokenNetwork(
+            custom_token.address,
+        ).transact(
+            {'from': B},
+        )
+
+    # Make sure deployer is administrator also for any created TokenNetwork
+    tx_hash = token_network_registry.functions.createERC20TokenNetwork(
+        custom_token.address,
+    ).transact(
+        {'from': administrator},
+    )
+    tx_receipt = web3.eth.getTransactionReceipt(tx_hash)
+    event_abi = contracts_manager.get_event_abi(
+        CONTRACT_TOKEN_NETWORK_REGISTRY,
+        EVENT_TOKEN_NETWORK_CREATED,
+    )
+    event_data = get_event_data(event_abi, tx_receipt['logs'][0])
+    token_network_address = event_data['args']['token_network_address']
+    token_network = web3.eth.contract(
+        abi=contracts_manager.get_contract_abi(CONTRACT_TOKEN_NETWORK),
+        address=token_network_address,
+    )
+
+    assert token_network.functions.administrator().call() == administrator
+
+
+def test_set_deprecation_switch(get_accounts, token_network):
+    (A) = get_accounts(1)[0]
+    administrator = token_network.functions.administrator().call()
+
+    assert token_network.functions.safety_deprecation_switch().call() is False
+
+    with pytest.raises(TransactionFailed):
+        token_network.functions.setSafetyDeprecationSwitch(False).transact({
+            'from': administrator,
+        })
+    with pytest.raises(TransactionFailed):
+        token_network.functions.setSafetyDeprecationSwitch(False).transact({
+            'from': A,
+        })
+
+    token_network.functions.setSafetyDeprecationSwitch(True).transact({
+        'from': administrator,
+    })
+    assert token_network.functions.safety_deprecation_switch().call() is True
+
+
+def test_deprecation_switch(get_accounts, token_network, create_channel, channel_deposit):
+    administrator = token_network.functions.administrator().call()
+    (A, B, C, D) = get_accounts(4)
+    deposit = 100
+    bigger_deposit = 200
+
+    channel_identifier = create_channel(A, B)[0]
+    channel_deposit(channel_identifier, A, deposit, B)
+    channel_deposit(channel_identifier, B, deposit, A)
+
+    token_network.functions.setSafetyDeprecationSwitch(True).transact({
+        'from': administrator,
+    })
+    assert token_network.functions.safety_deprecation_switch().call() is True
+
+    # Now we cannot deposit in existent channels
```

Comment:
nitpick: /s/existent/existing both here and in the rest of the tests below

[source ref](https://ell.stackexchange.com/a/161670)